experimental/dds/tree2/docs/undo/v1-undo.md

```diff
@@ -123,52 +127,410 @@ It can be still be thought of as sparse if we ignore the fact that sequenced com
+    (see diagrams below.)
 
-Repair data also needs to be (re)created when a local edit is rebased in the face of changes on the parent branch (possibly the trunk).
-In that case, we need to reapply the rebased version of the original edits to a separate forest solely for the purpose of generating the repair data.
-"Separate" here means that the forest is not the forest that drives updates to the application's UI.
+In all other cases, the repair data associated with E must be garbage-collected.
 
-The reason we need to use a separate forest is that the normal forest gets updated with a more efficient delta
-(which takes into account the fact that the original version of the rebased changes had already been applied).
-For example, if a local edit deleted some node foo,
-and that edit needed to be rebased over an incoming edit that edits foo in some way,
-then the forest that drives the UI should receive an empty `Delta`.
-However, for the purpose of getting the correct repair data for the rebased edits,
-we do need to get some forest into the state where foo exists and the incoming edit has been applied to it,
-so that we can gather the correct repair data for foo and associate that repair data with the rebased edit.
+This approach ensures that a client,
+no matter how long its history of undoable/redoable edits,
+does not burden its peers by requiring them to store unbounded amounts of repair data,
+and does not force summaries to include that repair data.
 
-There are two ways we can get a forest into that state:
+The one drawback,
+is that peers will not have the required repair data to apply the undo edit that a session might generate
+if the undo occurs sufficiently late.
+We address this problem by including in the undo commit being sent a copy of the relevant repair data,
+if the edit being undone lies outside of the current collaboration window.
+This acts as a "refresher" for any peers that may have garbage collected that repair data.
+(See [Creating Repair Data On Change Application](#creating-repair-data-on-change-application) for details on how that refresher is handled by the peers.)
 
-1. Fork the forest that drives the UI for that branch,
-   apply the inverse of the original (i.e., unrebased) local edits,
-   then apply the incoming edit(s).
-2. Fork the forest that drives the UI for the parent branch.
+If this switch from not including the refresher with the undo to including it came any later,
+meaning, if we did not include the refresher on undos created beyond the point where the undone edit falls out of the collaboration window,
+then we would run the risk of peers receiving an undo for which they have already discarded the repair data:
 
-Option #2 is bound to be more efficient but may require a more complex contract between parent and child branches.
-These options should be explored during implementation.
+![An undo edit is received by a peer that no longer has the repair data for it](./UndoAfterCollab.png)
```

Comment:
I don't quite understand what the dark blue collab window includes. Could you add an example?